js/cooknbook-javascript.js

```diff
@@ -1,81 +1,46 @@
 var mymap;
+//IT's always returning 401 unauthorized ?:?
+const EDMAM_ENDPOINT = `https://api.edamam.com/search?app_id=ae6bf7b0&app_key=f2c047350509b09ed7dc98cd15dd86d1—`;
+const FOURSQUARE_ENDPOINT = `https://api.foursquare.com/v2/venues/explore?client_id=IFQ0EC04QC55WLPD00E5XPI1MP03Z4UMQEHMQR34VSUQZS2C&client_secret=LNJVMGQ0EO2J20AYFNOZE2ROEIDJM4LPS1Y5ZMU244MXLNRZ&v=20180323&limit=5`
 
 let markerLayer;
 // Adding a click event listener to all elements with a class of "searchBtn"
-$(document).on("click", ".searchBtn", function(event) {
+$(document).on("click", ".searchBtn", async function (event) {
   event.preventDefault();
-
-  let query = $(event.target).attr("data-cuisine");
-
-  getRecipes(query);
-  getUserLocation(query);
+  try {
+    let query = $(event.target).attr("data-cuisine");
+    let userLocation = await getLocation();
+    var restaurants = await getRestaurants(query, userLocation);
+    renderMap(restaurants, userLocation);
+    const recipe = await getSingleRecipe(query);
+    renderRecipe(recipe)
+  } catch(err) {
+    console.log(err);
+  }
 });
 
 // get recipes from Edamam API
 
-function getRecipes(query) {
-  var queryURL =
-    "https://api.edamam.com/search?q=" +
-    query +
-    "&app_id=$ae6bf7b0&app_key=$f2c047350509b09ed7dc98cd15dd86d1—";
-  $.ajax({
-    url: queryURL,
+async function getSingleRecipe(query) {
+  return $.ajax({
```

Comment:
hi @tonysaad, can we not do
`const recipe = await $.ajax()
return recipe.hits[0].recipe` and get rid of the `then` altogether?